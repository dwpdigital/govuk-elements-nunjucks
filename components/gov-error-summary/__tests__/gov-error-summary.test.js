'use strict';

const nunjucks = require('nunjucks');

const errors = {'link-1': 'This is an error 1', 'link-2': 'This is an error 2'};

describe('gov-error-summary render', () => {
  it('should render h1, p and correct links when data is supplied', () => {
    const output = nunjucks.render('./components/gov-error-summary/template.njk', {
      heading: 'Error Heading',
      summary: 'Details of error',
      errors
    });
    expect(output).toMatchSnapshot();
  });
  it('should render without summary <p> if no summary is supplied', () => {
    const output = nunjucks.render('./components/gov-error-summary/template.njk', {
      heading: 'Error Heading',
      errors
    });
    expect(output).toMatchSnapshot();
  });
  it('should render nothing when no errors are supplied', () => {
    const output = nunjucks.render('./components/gov-error-summary/template.njk', {
      heading: 'Error Heading'
    });
    expect(output).toMatchSnapshot();
  });
});
