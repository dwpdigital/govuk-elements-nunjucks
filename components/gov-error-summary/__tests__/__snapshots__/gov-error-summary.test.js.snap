// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gov-error-summary render should render h1, p and correct links when data is supplied 1`] = `
<div class="error-summary" role="group" aria-labelledby="error-summary-heading"
tabindex="-1">
  <h1 class="heading-medium error-summary-heading" id="error-summary-heading">Error Heading</h1>
  <p>Details of error</p>
  <ul class="error-summary-list">
    <li>
      <a href="#link-1-form">This is an error 1</a>
    </li>
    <li>
      <a href="#link-2-form">This is an error 2</a>
    </li>
  </ul>
</div>
`;

exports[`gov-error-summary render should render nothing when no errors are supplied 1`] = `""`;

exports[`gov-error-summary render should render without summary <p> if no summary is supplied 1`] = `
<div class="error-summary" role="group" aria-labelledby="error-summary-heading"
tabindex="-1">
  <h1 class="heading-medium error-summary-heading" id="error-summary-heading">Error Heading</h1>
  <ul class="error-summary-list">
    <li>
      <a href="#link-1-form">This is an error 1</a>
    </li>
    <li>
      <a href="#link-2-form">This is an error 2</a>
    </li>
  </ul>
</div>
`;
